# HG changeset patch
# User Rik <rik@octave.org>
# Date 1414527349 25200
#      Tue Oct 28 13:15:49 2014 -0700
# Node ID 93a33123fcfe740611227ad7d8f1698df2dab1bf
# Parent  86f50893acd8a62f7c55797251dfba8e6844f6f2
# Parent  cd8cf41f130ddb3d2241054c2e637b632c8a21d6
maint: Periodic merge of gui-release to default.

diff --git a/liboctave/array/Range.h b/liboctave/array/Range.h
--- a/liboctave/array/Range.h
+++ b/liboctave/array/Range.h
@@ -49,17 +49,17 @@ public:
     : rng_base (b), rng_limit (l), rng_inc (i),
       rng_nelem (nelem_internal ()), cache () { }
 
   // For operators' usage (to preserve element count).
   Range (double b, double i, octave_idx_type n)
     : rng_base (b), rng_limit (b + (n-1) * i), rng_inc (i),
       rng_nelem (n), cache ()
   {
-    if (! xfinite (b) || ! xfinite (i) | ! xfinite (rng_limit))
+    if (! xfinite (b) || ! xfinite (i) || ! xfinite (rng_limit))
       rng_nelem = -2;
   }
 
   double base (void) const { return rng_base; }
   double limit (void) const { return rng_limit; }
   double inc (void) const { return rng_inc; }
   octave_idx_type nelem (void) const { return rng_nelem; }
 
